test/ETO/EquityToken.js

```diff
@@ -96,32 +99,113 @@ contract("EquityToken", ([admin, nominee, company, broker, ...holders]) => {
       expect(balance).to.be.bignumber.eq(initialBalance);
     });
 
-    it("should overflow on deposit");
+    it("should overflow on deposit", async () => {
+      const initialBalance = new web3.BigNumber(2).pow(256).plus(1);
+      await expect(
+        equityToken.issueTokens(initialBalance, {
+          from: company,
+        }),
+      ).to.be.rejectedWith(EvmError);
+    });
 
     // cases for successful destroy, rejected due to controller, not enough balance etc.
     it("should destroy tokens");
 
-    it("should change token controller");
-
-    it("should change nominee");
-
     // should be a set of tests with different rounding, we should be able to run it on platform as well
     it("should convert equity token amount to shares");
 
-    it("should set token symbol and other metadata from eto terms correctly");
+    it("should set token symbol and other metadata from eto terms correctly", async () => {
+      const equityTokenName = await etoTerms.EQUITY_TOKEN_NAME();
+      const equityTokenSymbol = await etoTerms.EQUITY_TOKEN_SYMBOL();
+      const equityTokenShareNominalValue = await etoTerms.SHARE_NOMINAL_VALUE_EUR_ULPS();
+      const equityTokenPrecision = await tokenTerms.EQUITY_TOKENS_PRECISION();
+      const equityTokensPerShare = await tokenTerms.EQUITY_TOKENS_PER_SHARE();
+
+      expect(await equityToken.name()).to.have.string(equityTokenName);
+      expect(await equityToken.symbol()).to.have.string(equityTokenSymbol);
+      expect(await equityToken.decimals()).to.be.bignumber.eq(equityTokenPrecision);
+      expect(await equityToken.tokensPerShare()).to.be.bignumber.eq(equityTokensPerShare);
+      expect(await equityToken.shareNominalValueEurUlps()).to.be.bignumber.eq(
+        equityTokenShareNominalValue,
+      );
+    });
   });
 
   describe("IEquityTokenController tests", () => {
-    // extend implementation of TestNullEquityTokenController to be able to control all operations via mocks
-    it("can block transfers");
-    it("can block distribute");
-    it("can block issueTokens");
-    it("can block destroyTokens");
-    it("can block closeToken");
-    it("can block changing equity token controller");
-    it("can block changing nominee");
-    it("should change token controller");
-    it("should change nominee");
+    it("should change token controller", async () => {
+      const newEquityTokenController = await TestNullEquityTokenController.new(universe.address);
+
+      await equityToken.changeTokenController(newEquityTokenController.address);
+
+      expect(await equityToken.tokenController()).to.be.bignumber.eq(
+        newEquityTokenController.address,
+      );
+
+      // clean up
+      await equityToken.changeTokenController(equityTokenController.address);
```

Comment:
you do not need to clean up after test. all tests start as clean slate - everything is redeployed.